embedding/embedlite/embedshared/EmbedLiteViewChild.cpp

```diff
@@ -979,13 +979,14 @@ mozilla::ipc::IPCResult EmbedLiteViewChild::RecvHandleDoubleTap(const LayoutDevi
   RefPtr<Document> document = presShell->GetDocument();
   NS_ENSURE_TRUE(document && !document->Fullscreen(), IPC_OK());
 
-  CSSRect zoomToRect = CalculateRectToZoomTo(document, cssPoint);
-
-  uint32_t presShellId;
-  ViewID viewId;
-  if (APZCCallbackHelper::GetOrCreateScrollIdentifiers(
-      document->GetDocumentElement(), &presShellId, &viewId)) {
-    ZoomToRect(presShellId, viewId, zoomToRect);
+  if (nsLayoutUtils::AllowZoomingForDocument(document)) {
+    CSSRect zoomToRect = CalculateRectToZoomTo(document, cssPoint);
+    uint32_t presShellId;
+    ViewID viewId;
+    if (APZCCallbackHelper::GetOrCreateScrollIdentifiers(
+        document->GetDocumentElement(), &presShellId, &viewId)) {
+      ZoomToRect(presShellId, viewId, zoomToRect);
+    }
   }
 
   if (sPostAZPCAsJson.doubleTap) {
```

Comment:
You should enable this to the embedding.js in order this pipeline to work. Currently, it is not enabled.

When enabled Gesture:DoubleTap starts interfering with doubletaptozoom. For instance visible in the testinputfocus.html (zooming allowed) when do following:

1) Open test case
2) Pinch in so that last three paragraphs do fit to the screen
3) Double tap the last paragraph

Expected outcome:
Page is fully zoomed out

Actual outcome:
Page is fully zoomed out and text gets selected

Additional info:
Easier to reproduce with a page that does not have contentEditable set.

Reply:
Would be good to move defaults that we have in webenginesettings to the embedding.js